packages/ui-causal-graph-editor/src/shared/use-causal-graph-editor.tsx

```diff
@@ -63,35 +96,35 @@ export default function useCausalGraphEditor(
         }
     );
 
+    // bind each action creator to dispatch
+    const api = useMemo(() => {
+        return actionNames.reduce<GraphApi>((acc, actionName) => {
+            const actionCreator = GraphActionCreators[actionName];
+            // eslint-disable-next-line prefer-spread
+            acc[actionName] = (...args: Parameters<typeof actionCreator>) => dispatch(actionCreator.apply(null, args));
+            return acc;
+        }, {} as GraphApi);
+    }, [dispatch]);
+
     const isTimeSeriesCausalGraph = useMemo(() => {
         const nodeClass = state.graph.getNodeAttribute(state.graph.nodes()[0], 'extras')?.node_class;
         return nodeClass === 'TimeSeriesNode';
     }, [state.graph]);
 
     const layout = useMemo(() => {
-        // If we have a time series causasl graph and if the layout chosen supports tiers and these are not defined
         const newLayout = graphLayout;
+
         if (
             isTimeSeriesCausalGraph &&
             'tiers' in newLayout &&
             'orientation' in newLayout &&
             newLayout.tiers === undefined
         ) {
-            // We set tiers based on TimeSeriesCausalGraph structure of node lags
-            newLayout.tiers = { group: 'variable_name', order_nodes_by: 'time_lag' };
+            updateNodesForTimeSeries(state.graph, api);
```

Comment:
It's an anti-pattern to fire side effects within a `useMemo`, can we think of a better way to do it?

Reply:
After reading the useEffect page you suggested, from what I understood their recommended approach is to be outside of a memo.